kratos/solving_strategies/convergencecriterias/displacement_criteria.h

```diff
@@ -193,13 +209,13 @@ class DisplacementCriteria
 
             const TDataType absolute_norm = (final_correction_norm/std::sqrt(float_size_solution));
 
-            KRATOS_INFO_IF("DISPLACEMENT CRITERION", this->GetEchoLevel() > 0 && rModelPart.GetCommunicator().MyPID() == 0) << " :: [ Obtained ratio = " << ratio << "; Expected ratio = " << mRatioTolerance << "; Absolute norm = " << absolute_norm << "; Expected norm =  " << mAlwaysConvergedNorm << "]" << std::endl;
+            KRATOS_INFO_IF("DISPLACEMENT CRITERION", this->GetEchoLevel() > 0 && rank == 0) << " :: [ Obtained ratio = " << ratio << "; Expected ratio = " << mRatioTolerance << "; Absolute norm = " << absolute_norm << "; Expected norm =  " << mAlwaysConvergedNorm << "]" << std::endl;
 
             rModelPart.GetProcessInfo()[CONVERGENCE_RATIO] = ratio;
             rModelPart.GetProcessInfo()[RESIDUAL_NORM] = absolute_norm;
 
             if ( ratio <= mRatioTolerance  ||  absolute_norm<mAlwaysConvergedNorm )  { //  || (final_correction_norm/x.size())<=1e-7)
-                KRATOS_INFO_IF("DISPLACEMENT CRITERION", this->GetEchoLevel() > 0 && rModelPart.GetCommunicator().MyPID() == 0) << "Convergence is achieved" << std::endl;
+                KRATOS_INFO_IF("DISPLACEMENT CRITERION", this->GetEchoLevel() > 0 && rank == 0) << "Convergence is achieved" << std::endl;
                 return true;
             } else {
                 return false;
```

Comment:
Not something you have changed, but since you are refactoring code I believe its better to change all this blocks to:

```c++
bool hasAchievedConvergence = ratio <= mRatioTolerance  ||  absolute_norm<mAlwaysConvergedNorm;

KRATOS_INFO_IF("DISPLACEMENT CRITERION", hasAchievedConvergence && this->GetEchoLevel() > 0 && rank == 0) << "Convergence is achieved" << std::endl;

return hasAchievedConvergence;
```

Reply:
Okay